src/AutomatedCar/SystemComponents/CarCollisionDetector.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace AutomatedCar.SystemComponents
+{
+    using AutomatedCar.Models;
+    using AutomatedCar.SystemComponents.Packets;
+    using Avalonia;
+    using Avalonia.Media;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    public class CarCollisionDetector : SystemComponent
+    {
+        private CollisionPacket packet;
+        private IEnumerable<WorldObject> collidableWorldObjects;
+
+        public CarCollisionDetector(VirtualFunctionBus virtualFunctionBus)
+            : base(virtualFunctionBus)
+        {
+            this.packet = new CollisionPacket();
+            this.collidableWorldObjects = World.Instance.WorldObjects
+                                        .Where(obj => obj.Collideable
+                                        && obj != World.Instance.ControlledCar);
+        }
+
+        public override void Process()
+        {
+            var collisions = this.collidableWorldObjects
+                            .Where(obj => CollisionCheck(World.Instance.ControlledCar, obj));
+
+            var dynamicWorldObjects = collisions
+                           .Where(obj => obj.WorldObjectType == WorldObjectType.Car
```

Comment:
Indent 1 tab only. Instead of using `==` twice use it in a pattern like `obj => obj.WorldObjectType is WorldObjectType.Car or WorldObjectType.Pedestrian`
Do this at all occurances with the indenting aswell.